File: src/main/java/mediathek/gui/actions/StartAllDownloadsAction.java
package mediathek.gui.actions;

import mediathek.gui.tabs.tab_downloads.GuiDownloads;
import mediathek.swing.IconUtils;
import org.kordamp.ikonli.fontawesome6.FontAwesomeSolid;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class StartAllDownloadsAction extends AbstractAction {
    private final GuiDownloads guiDownloads;

    public StartAllDownloadsAction(GuiDownloads guiDownloads) {
        this.guiDownloads = guiDownloads;
        putValue(Action.SMALL_ICON, IconUtils.toolbarIcon(FontAwesomeSolid.ANGLE_DOUBLE_DOWN));
        putValue(Action.SHORT_DESCRIPTION, "Alle Downloads starten");
        putValue(Action.NAME, "Alle Downloads starten");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        guiDownloads.starten(true);
    }
}

File: src/main/kotlin/mediathek/gui/actions/StartAllDownloadsAction.kt
package mediathek.gui.actions

import mediathek.gui.tabs.tab_downloads.GuiDownloads
import mediathek.swing.IconUtils
import org.kordamp.ikonli.fontawesome6.FontAwesomeSolid
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class StartAllDownloadsAction(private val guiDownloads: GuiDownloads) : AbstractAction() {
    init {
        putValue(SMALL_ICON, IconUtils.toolbarIcon(FontAwesomeSolid.ANGLE_DOUBLE_DOWN))
        putValue(SHORT_DESCRIPTION, "Alle Downloads starten")
        putValue(NAME, "Alle Downloads starten")
    }

    override fun actionPerformed(e: ActionEvent?) {
        guiDownloads.starten(true)
    }
}
